src/tracer.js

```diff
@@ -254,23 +260,19 @@ export default class Tracer {
     }
 
     /**
-     *  Makes a concrete sampling decision for the ctx span context based on information
-     *  available to it's child span.
-     *
-     *  The ctx context's deferred mask is unset after this decision is made
+     * Reapplies sampling if the parent span isn't sampled
      *
      * @param ctx the span context
      * @param operationName the operation name for the child
      * @param tags tags to be applied by the sampler
      */
-    processDeferredSampling(ctx: SpanContext, operationName: string, tags: any) {
-        if (ctx.isDeferredSampling) {
-            if (this._sampler.isSampled(operationName, tags)) {
-                ctx._flags |= constants.SAMPLED_MASK;
-            } else {
-                ctx._flags &= ~constants.SAMPLED_MASK;
-            }
-            ctx.unsetDeferredSampling();
+    handleUpsampling(ctx: SpanContext, operationName: string, tags: any) {
```

Comment:
I think it is, otherwise you are going to keep re-sampling on every inner span. By default we only want to re-sample on the first in process span